src/content/docs/en/recipes/minification.mdx

```diff
@@ -0,0 +1,61 @@
+---
+title: Minify HTML responses with middleware
+description: Use Astro middleware to minify HTML responses
+i18nReady: false
+type: recipe
+---
+
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+
+Astro supports built-in [HTML minification](/en/reference/configuration-reference/#compresshtml) which works for `.astro` components, but not for [islands](/en/concepts/islands/). This recipe shows how to use Astro middleware to minify complete HTML responses.
+
+## Recipe
+
+1. Install the `html-minifier` package:
+
+  <PackageManagerTabs>
+    <Fragment slot="npm">
+    ```shell
+    npm install html-minifier
+    ```
+    </Fragment>
+    <Fragment slot="pnpm">
+    ```shell
+    pnpm install html-minifier
+    ```
+    </Fragment>
+    <Fragment slot="yarn">
+    ```shell
+    yarn add html-minifier
+    ```
+    </Fragment>
+  </PackageManagerTabs>
+
+2. Create a middleware file `src/middleware.ts`:
```

Comment:
```suggestion
2. Create a middleware file `src/middleware.ts`. Add the following code to intercept an HTML response and returned a minified version:
```
I think a *bit* of explanation here, just at this level, is helpful! Nothing more than would comfortably fit in a sentence. Adjust so that it's accurate!